packages/rum-core/src/common/patching/event-target-patch.js

```diff
@@ -42,7 +42,7 @@ for (let i = 0; i < eventTypes.length; i++) {
 
 function shouldInstrumentEvent(target, eventType, listenerFn) {
   return (
-    target instanceof Element &&
+    (target instanceof Element || target === window || target === document) &&
```

Comment:
this would effectively mean we start capturing all click events everywhere on the page which are using event bubbling (Ex: react). As a side effect we would get everything as `Click - document` which is not meaningful. One way to fix this problem is to use the `target.activeElement` if available on the taret. https://developer.mozilla.org/en-US/docs/Web/API/Document/activeElement

This would work for all of the cases mentioned here - https://github.com/elastic/apm-agent-rum-js/issues/969#issue-812096856

Reply:
But React 17 Listen on `div#root`.
https://reactjs.org/blog/2020/08/10/react-v17-rc.html#changes-to-event-delegation

Reply:
My suggestion:

```javascript
if (target === window || target === document ||
    target && isFunction(target.contains) && target.contains(document.activeElement)) {
  // use document.activeElement instead of target
}
```

https://developer.mozilla.org/en-US/docs/Web/API/Node/contains

Reply:
Or we can always use `document.activeElement` instead of `target`.

Anyway, it's a breaking change that using `document.activeElement` when target is not `document` or `window`.

Reply:
Good point about the react 17 breaking change. So instead of touching the code here, Let's keep it as it is and check for target being on the `document`. Instead check when the event is actually called https://github.com/elastic/apm-agent-rum-js/blob/8dda5c57a55f815f3252f7e2e769191efeefd7fe/packages/rum-core/src/performance-monitoring/performance-monitoring.js#L190

Check if the target contains any `activeElement`. 

If you can do a test on that would be great. I will take your PR for a spin today and test it out.

Reply:
I've tested on React 17, and it works.

```
[Elastic APM] startTransaction(43fbc8c4fe6af7f8, Click - button["button"], user-interaction)
```